### DISCUSSION TITLE
GC-NSCBC method

### DISCUSSION THREAD

**Discussion by czc-zju** [AMReX-Combustion/PeleC#455] [General]
https://pelec.readthedocs.io/en/latest/BoundaryConditions.html 
![image](https://user-images.githubusercontent.com/81547457/159877050-e1309a60-d913-49f9-8777-e5efb5e672c7.png)，

when I impose a "Hard" BC for the inlet and a "FOExtrap" for the outlet,  the fluid always is reflected from the outflow boundary back into the domain, how should I avoid it?

**Comment by drummerdoc** [comment]
Are you sure that it is not just the acoustic wave that is reflected?

**Comment by marchdf** [comment]
Hi, that part of the documentation is out of date. I apologize for that. NSCBC capability has not been migrated to the GPU-capable code yet. Currently, if the "Hard" BC is chosen, the user must define the BC in the`bcnormal` function in `prob.H`. This can be done as an inflow or an outflow. "FOExtrap" can be used as an outflow BC but acoustic waves may be reflected. 

The flows you are showing seem to indicate something else is wrong. Potentially in the input files. If you share your case files, we may be able to help diagnose the issue.

**Reply by marchdf** [reply]
Actually the documentation is not out of date... But RTD is somehow not picking up the updates. This is what it should be showing:
![Screen Shot 2022-03-24 at 8 11 15 AM](https://user-images.githubusercontent.com/15038415/159935010-8a8def31-5bc0-48b2-972d-486b2b62da6b.png)

@jrood-nrel do you know why the RTD is not using the right docs?

**Reply by marchdf** [reply]
Ok nvm I figured it out. The right url to be using for the documentation is: [https://amrex-combustion.github.io/PeleC/BoundaryConditions.html]( https://amrex-combustion.github.io/PeleC/BoundaryConditions.html). It's not clear to me where that other RTD link posted at the top came from.

**Reply by czc-zju** [reply]
I already sent you the e-mail (_marchdf@umich.edu_) about my case files and I hope to get your help.

**Reply by marchdf** [reply]
I think that email address is defunct. Can you post your files here? It's best for future users as well. If you don't want to share your files publicly, we can find another way.

**Comment by marchdf** [comment]
I will note that if one wants to do a better job at outflow than FOExtrap (without doing NSCBC), one can follow this [BC implementation](https://github.com/AMReX-Combustion/PeleC/blob/6bb0d7b1622be9e80dfb8d2d2999c72934b19ea6/Exec/RegTests/EB-C10/prob.H#L94).

**Reply by czc-zju** [reply]
Thank you for sharing, they are very helpful for me.
I'll try it now.

**Comment by czc-zju** [comment]


**Reply by marchdf** [reply]
I am sorry but things are looking a little garbled here and I don't see any attachements. Could you use the GitHub interface here: https://github.com/AMReX-Combustion/PeleC/discussions/455
